### PR TITLE
(Re)PR for remaining changes for (PR #7643)

### PR DESCRIPTION
**Pull request by NitiKaur**
info abt hardware SPI Buses added, changes made to the ADC module, info about watchdog class corrected

### CODE REVIEW

File: docs/rp2/quickref.rst [hunk 1]
```diff
@@ -112,14 +142,17 @@ Use the ``machine.PWM`` class::
 ADC (analog to digital conversion)
 ----------------------------------
 
-How does the ADC module work?
+RP2040 has five ADC channels in total, four of which are 12-bit SAR based
+ADCs: GP26, GP27, GP28 and GP29. The input signal for ADC0, ADC1 and ADC2 can
+be connected with GP26, GP27, GP28. The standard ADC range is 0-3.3V. The fifth channel
```
Comment (dpgeorge) [inline docs/rp2/quickref.rst:147]: I think it's possible to use ADC3 with GP29.  Or did you find information that says it won't work?

File: docs/rp2/quickref.rst [hunk 2]
```diff
@@ -84,19 +97,36 @@ Use the :ref:`machine.Pin <machine.Pin>` class::
 UART (serial bus)
 -----------------
 
+There are two UARTs, UART0 and UART1. UART0 can be mapped to GPIO 1/2, 12/13
+and 16/17, and UART1 to GPIO 6/7 and 8/9. Since for using the UART GPIO
+numbers have to be used, these should be mentioned and can be seen in the pin
+diagram below:
+
+.. image:: img/UARTpinout.png
```
Comment (dpgeorge) [inline docs/rp2/quickref.rst:105]: This image has not been added as part of this PR, so the docs have an error when building.

What I would suggest is to change the image at the top of the quickref to the full Pico pinout, eg the one here: https://peppe8o.com/raspberry-pi-pico-pinout/

File: docs/rp2/quickref.rst [hunk 3]
```diff
@@ -27,10 +27,9 @@ a troubleshooting subsection.
 General board control
 ---------------------
 
-The MicroPython REPL is on the USB serial port.
-Tab-completion is useful to find out what methods an object has.
-Paste mode (ctrl-E) is useful to paste a large slab of Python code into
-the REPL.
+The MicroPython REPL is on the USB serial port. Tab-completion is useful to
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:30]:
```suggestion
The MicroPython REPL is accessed via the USB serial port. Tab-completion is useful to
```

File: docs/rp2/quickref.rst [hunk 4]
```diff
@@ -59,7 +58,21 @@ Use the :mod:`time <utime>` module::
 Timers
 ------
 
-How do they work?
+RP2040's system timer peripheral provides a global microsecond timebase and
+generates interrupts for the same. The software timer is available currently,
+and there are unlimited number of them available (limited by memory). There is
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:63]:
```suggestion
and there are unlimited number of them available (memory permitting). There is
```

File: docs/rp2/quickref.rst [hunk 5]
```diff
@@ -84,19 +97,36 @@ Use the :ref:`machine.Pin <machine.Pin>` class::
...
+
+.. image:: img/UARTpinout.png
+    :alt: Pin Diagram Raspberry Pi Pico
+    :width: 640px
+
 See :ref:`machine.UART <machine.UART>`. ::
 
     from machine import UART
 
     uart1 = UART(1, baudrate=9600, tx=33, rx=32)
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:113]: Should these lines be removed?

File: docs/rp2/quickref.rst [hunk 6]
```diff
@@ -84,19 +97,36 @@ Use the :ref:`machine.Pin <machine.Pin>` class::
...
+    REPL over UART is disabled by default. You can see the :ref:`rp2_intro` for
+    details on how to enable REPL over UART.
+
 
 PWM (pulse width modulation)
 ----------------------------
 
-How does PWM work on the RPi RP2xxx?
+There are 8 independent channels each of which have 2 outputs making it 16
+PWM channels in total which can be clocked from 7Hz to 125Mhz.
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:129]: Question (perhaps for @dpgeorge): should we specify the PWM resolution?

File: docs/rp2/quickref.rst [hunk 7]
```diff
@@ -156,14 +189,15 @@ Software SPI (using bit-banging) works on all pins, and is accessed via the
...
+The RP2040 has 2 hardware SPI buses which is accessed via the
+:ref:`machine.SPI <machine.SPI>` class and has the same methods as software
+SPI above::
 
     from machine import Pin, SPI
 
-    spi = SPI(1, 10000000)
-    spi = SPI(1, 10000000, sck=Pin(14), mosi=Pin(13), miso=Pin(12))
-    spi = SPI(2, baudrate=80000000, polarity=0, phase=0, bits=8, firstbit=0, sck=Pin(18), mosi=Pin(23), miso=Pin(19))
+    spi = SPI(1, 10000000)  # Default assignment: sck=Pin(10), mosi=Pin(11), miso=Pin(8)
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:198]: Just a suggestion, but MicroPython supports underscores in numeric literals which improves readability particularly with large numbers:

```suggestion
    spi = SPI(1, 10_000_000)  # Default assignment: sck=Pin(10), mosi=Pin(11), miso=Pin(8) 
```

File: docs/rp2/quickref.rst [hunk 8]
```diff
@@ -202,13 +236,14 @@ See :ref:`machine.RTC <machine.RTC>` ::
     from machine import RTC
 
     rtc = RTC()
-    rtc.datetime((2017, 8, 23, 2, 12, 48, 0, 0)) # set a specific date and time
+    rtc.datetime((2017, 8, 23, 1, 12, 48, 0, 0)) # set a specific date and time
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:239]: Perhaps add that datetime to the end of the comment: "2017/8/23 1:12:48"

File: docs/rp2/quickref.rst (hunk 6 above)
Comment (dpgeorge) [reply docs/rp2/quickref.rst:129]: Not at this stage.

File: docs/rp2/quickref.rst (hunk 1 above)
Comment (dpgeorge) [reply docs/rp2/quickref.rst:147]: Include ADC3/GP29 in this list but add a note that "on Pico GP29 is connected to VSYS"

**Comment by dpgeorge** [comment]
Thanks for updating this PR.  I squashed the commits and merged in da74ef661587011132570abfe70114ca8ccd679c with some minor fixes.  I also changed the image to a better pinout of the Pico.

Good work!

**Pull request closed without merging**